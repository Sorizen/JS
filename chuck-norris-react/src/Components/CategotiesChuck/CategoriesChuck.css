.categories-wrapper{
    width: 1120px;
    margin-right: auto;
    margin-left: auto;
    margin-top: 60px;
}
.categories-wrapper h1{
    text-align: center;
    font-size: 50px;
    line-height: 58px;
}
.category-items-wrapper{
    margin-top: 60px;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    align-items: center;
}
.category-item{
    display: flex;
    align-items: center;
    justify-content: center;
    border-radius: 18px;
    width: 170px;
    height: 57px;
    font-size: 16px;
    line-height:18px;
    background-color: white;
    transition: all 0.2s ease-in-out;
    cursor: pointer;
    margin-bottom: 20px;
    filter: drop-shadow(0px 22px 80px rgba(72, 70, 106, 0.07)) drop-shadow(0px 3px 11px rgba(72, 70, 106, 0.035));
}
.category-item.category-item-margin-left{
    margin-left: 95px;
}
.category-item.category-item-margin-right{
    margin-right: 95px;
}
.category-item:hover{
    color: white;
    background-color: #422ED4;
    filter: drop-shadow(0px 22px 80px rgba(72, 70, 106, 0.07)) drop-shadow(0px 3px 11px rgba(72, 70, 106, 0.035));
}
@media screen and (max-width: 1240px){
    .categories-wrapper{
        width: 1000px;
    }
    .category-item{
        width: 165px;
        font-size: 14px;
    }
    .category-item.category-item-margin-left{
        margin-left: 37px;
    }
    .category-item.category-item-margin-right{
        margin-right: 37px;
    }
}
@media screen and (max-width: 1024px){
    .categories-wrapper{
        width: 700px;
    }
    .category-item{
        width: 165px;
        font-size: 14px;
    }
    .category-item.category-item-margin-left{
        margin-left: 0;
    }
    .category-item.category-item-margin-right{
        margin-right: auto;
        margin-left: auto;
    }
}
@media screen and (max-width: 768px){
    .categories-wrapper{
        width: 400px;
        justify-content: space-around;
    }
    .category-items-wrapper{
        justify-content: space-around;
    }
    .category-item{
        width: 165px;
        font-size: 14px;
    }
    .category-item.category-item-margin-left{
        margin-left: 0;
    }
    .category-item.category-item-margin-right{
        margin-right: auto;
        margin-left: auto;
    }
}
@media screen and (max-width: 425px){
    .categories-wrapper{
        width: 300px;
        justify-content: space-around;
    }
    .category-items-wrapper{
        justify-content: space-around;
    }
    .categories-wrapper h1{
        font-size: 40px;
    }
    .category-item{
        width: 135px;
        font-size: 14px;
    }
    .category-item.category-item-margin-left{
        margin-left: 0;
    }
    .category-item.category-item-margin-right{
        margin-right: auto;
        margin-left: auto;
    }
}