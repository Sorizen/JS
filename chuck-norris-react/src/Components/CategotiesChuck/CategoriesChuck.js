import React from "react";
import './CategoriesChuck.css';
const CategoriesChuck = (props) =>{
    let classNameForItem = ["category-item"];
    return(
        <div className="categories-wrapper">
            <h1>Categories</h1>
            <div className="category-items-wrapper">
                {props.categories ? props.categories.map((category, index)=>{
                    if(category === 'religion'){
                        classNameForItem.push("category-item-margin-left");
                    }
                    else if (category === 'random'){
                        classNameForItem.push("category-item-margin-right");
                    }
                    else{
                        classNameForItem = ["category-item"];
                    }
                    return(
                        <div className={classNameForItem.join(' ')} key={index} onClick={()=>{
                            props.setSwitcher(!props.Switcher);
                            props.setActiveCategory(category);
                        }}>
                            {category}
                        </div>
                    )
                }): null}
            </div>
        </div>
    )
}
export default CategoriesChuck