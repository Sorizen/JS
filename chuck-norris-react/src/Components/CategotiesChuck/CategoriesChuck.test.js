import {render, screen} from '@testing-library/react';
import CategoriesChuck from "./CategoriesChuck";
const ChuckCategoriesTest = ['animal', 'random', 'unit'];
describe('CategoriesChuck component', ()=>{
    test('CategoriesChuck renders', ()=>{
        render(<CategoriesChuck categories={ChuckCategoriesTest}/>);
        expect(screen.getByText(ChuckCategoriesTest[0])).toBeInTheDocument();
        expect(screen.getByText(ChuckCategoriesTest[1])).toBeInTheDocument();
        expect(screen.getByText(ChuckCategoriesTest[2])).toBeInTheDocument();
    })
    test('CategoriesChuck renders without props', ()=>{
        render(<CategoriesChuck/>);
        expect(screen.queryByText(ChuckCategoriesTest[0])).toBeNull();
        expect(screen.queryByText(ChuckCategoriesTest[1])).toBeNull();
        expect(screen.queryByText(ChuckCategoriesTest[2])).toBeNull();
    })
    test('CategoriesChuck snapshot with props', ()=>{
        const CategoriesChuckPrev = render(<CategoriesChuck categories={ChuckCategoriesTest}/>);
        expect(CategoriesChuckPrev).toMatchSnapshot();
    })
    test('CategoriesChuck snapshot without props', ()=>{
        const CategoriesChuckPrev = render(<CategoriesChuck/>);
        expect(CategoriesChuckPrev).toMatchSnapshot();
    })
})
