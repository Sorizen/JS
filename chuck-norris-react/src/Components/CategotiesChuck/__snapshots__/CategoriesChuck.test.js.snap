// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesChuck component CategoriesChuck snapshot with props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="categories-wrapper"
      >
        <h1>
          Categories
        </h1>
        <div
          class="category-items-wrapper"
        >
          <div
            class="category-item"
          >
            animal
          </div>
          <div
            class="category-item category-item-margin-right"
          >
            random
          </div>
          <div
            class="category-item"
          >
            unit
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="categories-wrapper"
    >
      <h1>
        Categories
      </h1>
      <div
        class="category-items-wrapper"
      >
        <div
          class="category-item"
        >
          animal
        </div>
        <div
          class="category-item category-item-margin-right"
        >
          random
        </div>
        <div
          class="category-item"
        >
          unit
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CategoriesChuck component CategoriesChuck snapshot without props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="categories-wrapper"
      >
        <h1>
          Categories
        </h1>
        <div
          class="category-items-wrapper"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="categories-wrapper"
    >
      <h1>
        Categories
      </h1>
      <div
        class="category-items-wrapper"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
